examples/openai/README.md

```diff
@@ -1,7 +1,298 @@
+      prompt: `Suggest an item for the menu of a ${subject} themed restaurant`,
+      model: gpt4Turbo,
+    });
+
+    return llmResponse.text();
+  }
+);
+
+// Run the flow
+const response = await runFlow(myFlow, 'mexican' );
+console.log(await response.text());
+```
+
+### Embedders and retrievers
+
+Apart from the text generation models, Genkit also features the access to the text embeddings models and 
+implements retrievers which can retrieve documents, given a query. To use the text embedding models,
+you should utilize the `embed` method:
+```
+import { textEmbedding3Small } from 'genkitx-openai-plugin'; 
+import { embed } from '@genkit-ai/ai/embedder';
+
+const embedding = embed({
+  embedder: textEmbedding3Small,
+  content: "Embed this text.",
+});
+```
+Here, the variable `embedding` will be a vector of numbers, which is a latent space representation of the given text, which can find use in many downstream tasks. In this case, we can use the text embeddings in a retriever, to query similar documents from Firestore based on the extracted embeddings:
+```
+import { embed } from '@genkit-ai/ai/embedder';
+import { Document, defineRetriever } from '@genkit-ai/ai/retriever';
+import { textEmbedding3Small } from 'genkitx-openai-plugin';
+import {
+  FieldValue,
+  VectorQuery,
+  VectorQuerySnapshot,
+} from '@google-cloud/firestore';
+import { Firestore } from 'firebase-admin/firestore';
+import * as z from 'zod';
+
+const QueryOptions = z.object({
+  k: z.number().optional(),
+});
+
+const firestoreArtifactsRetriever = defineRetriever(
+  {
+    name: 'firestore/artifacts',
+    configSchema: QueryOptions,
+  },
+  async (input, options) => {
+    const embedding = await embed({
+      embedder: textEmbedding3Small,
+      content: input,
+    });
+
+    const db = new Firestore();
+    const coll = db.collection('vectors' /* your collection name */);
+
+    const vectorQuery: VectorQuery = coll.findNearest(
+      'embedding' /* the name of the field that contains the vector */,
+      FieldValue.vector(embedding),
+      {
+        limit: options.k ?? 3,
+        distanceMeasure: 'COSINE',
+      }
+    );
+
+    const vectorQuerySnapshot: VectorQuerySnapshot = await vectorQuery.get();
+    return {
+      documents: vectorQuerySnapshot.docs.map((doc) =>
+        // doc.data() represents the Firestore document. You may process
+        // it as needed to generate a Genkit document object, depending on your
+        // storage format.
+        Document.fromText(doc.data().content.text)
+      ),
+    };
+  }
+);
+```
+In the [official Genkit documentation](Add Genkit docs link) you can find many more usage examples as well as detailed specification and full instructions of the usage of the Genkit framework.
 
 ## Contributing
 If you want to contribute to the OpenAI plugin, you can link to a local instance of the plugin by running `npm link` in the plugin directory and `npm link genkitx-openai-plugin` in the examples directory.
```

Comment:
in order to prevent this from getting buried, I'd consider putting the usage examples in a separate doc that gets linked in the examples/openai/README.md file

This will also be useful if we move to a structure where we have examples for all plugins in one package

Reply:
I couldn't find the markdown support for it, only some external tools that allow that. If I got you right, you want the same content, but would like to separate the usage examples in the new markdown file which gets imported in our README.md?

Reply:
Or to omit the examples in the main README and just link to other .md file which will contain them?

Reply:
Why not: [Usage examples OpenAI Plugin](docs/usage_openai.md)?
I'd be in favor of the latter. so we can link to docs/usage_openai.md and other ones. lmk if this makes sense